CONTRIBUTING.md

```diff
@@ -19,15 +31,131 @@ pnpm lint
+When defining conditions for decryption, context parameters are specified to indicate what information is required.
+These parameters are placeholders that will be filled with actual data during the decryption process.
+
+The built in `:userAddress` parameter is defined in `packages/taco-auth/src/auth-provider.ts`.
+Custom context parameters can also be defined and they must always begin with `:`.
+
+[This test](https://github.com/nucypher/taco-web/blob/b689493a37bec0b168f80f43347818095c3dd5ce/packages/taco/test/taco.test.ts#L102) demonstrates the functionality:
+```typescript
+  it('exposes requested parameters', async () => {
+    const mockedDkg = fakeDkgFlow(FerveoVariant.precomputed, 0, 4, 4);
+    const mockedDkgRitual = fakeDkgRitual(mockedDkg);
+    const provider = fakeProvider(aliceSecretKeyBytes);
+    const signer = fakeSigner(aliceSecretKeyBytes);
+    const getFinalizedRitualSpy = mockGetActiveRitual(mockedDkgRitual);
+
+    const customParamKey = ':nftId';
+    const ownsNFTWithCustomParams = new conditions.predefined.erc721.ERC721Ownership({
+      contractAddress: '0x1e988ba4692e52Bc50b375bcC8585b95c48AaD77',
+      parameters: [customParamKey],
+      chain: TEST_CHAIN_ID,
+    });
+
+    const messageKit = await taco.encrypt(
+      provider,
+      domains.DEVNET,
+      message,
+      ownsNFTWithCustomParams,
+      mockedDkg.ritualId,
+      signer,
+    );
+    expect(getFinalizedRitualSpy).toHaveBeenCalled();
+
+    const requestedParameters = taco.conditions.context.ConditionContext.requestedContextParameters(messageKit);
+    expect(requestedParameters).toEqual(new Set([customParamKey, USER_ADDRESS_PARAM_DEFAULT]));
+  });
+```
+
+## Authentication Providers
+User authentication is handled by an `AuthProvider` defined in `packages/taco-auth/src`.
+Using authentication providers has several benefits:
+1. Abstraction:
+
+    They abstract away the specific details of authentication, providing a unified interface for different authentication methods.
+
+2. Flexibility:
+
+    The system can accommodate various types of authentication providers, making it extensible and adaptable to different authentication needs.
+
+3. User Context Management:
+
+    They help manage user context, specifically the `:userAddress` parameter. This includes handling authentication tokens and ensuring that users are authenticated only once for multiple actions.
+
+Currently, [SIWE](https://docs.login.xyz/) (Sign In With Ethereum, [EIP-4361](https://eips.ethereum.org/EIPS/eip-4361)) is the only authentication method implemented.
+EIP-712 has previously been supported but is now deprecated.
+
+[The below test](https://github.com/nucypher/taco-web/blob/b689493a37bec0b168f80f43347818095c3dd5ce/packages/taco/test/conditions/context.test.ts#L382C1-L429C6) demonstrates how a SIWE message can be reused for TACo authentication.
+This ensures that users don't have to sign multiple messages when logging into apps and decrypting TACo messages.
```

Comment:
I would put it this way, WDYT?

```suggestion
Reusing a SIWE authentication message allows the users to log in to the Dapp so that this authentication is also used for TACo decryption, i.e. users only have to sign authentication messages once.
```